erizo/src/erizo/rtp/QualityFilterHandler.cpp

```diff
@@ -120,6 +127,23 @@ void QualityFilterHandler::write(Context *ctx, std::shared_ptr<DataPacket> packe
   if (is_scalable_ && !chead->isRtcp() && enabled_ && packet->type == VIDEO_PACKET) {
     RtpHeader *rtp_header = reinterpret_cast<RtpHeader*>(packet->data);
 
+    if (cooldown && clock::now() > cooldown_target) {
+    	cooldown = false;
+    	ELOG_DEBUG("End cooldown")
+    }
+
+    // Force switch to lower resolution if the packet frequency is > kSwitchDownTimeout (ms)
+    if (!cooldown && clock::now() - last_sent_packet > kSwitchDownTimeout) {
```

Comment:
can we move this  if clause to `changeSpatialLayerOnKeyframeReceived()`? I think it makes more sense to have it there.